code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -371,3 +371,38 @@
 	desc = "An upgraded model of the basic laser gun. There seems to be some sort of slot in the handle."
 	can_take_pai = TRUE
 	origin_tech = Tc_COMBAT + "=3;" + Tc_MAGNETS + "=2;" + Tc_ENGINEERING + "=2;" + Tc_PROGRAMMING + "=4"
+
+/obj/item/weapon/gun/energy/laser/rainbow
+
+	name = "rainbow laser"
+	desc = "A fearsome gun used by clown special forces"
+	projectile_type = "/obj/item/projectile/beam/rainbow"
+	var/current_color = 1
+	var/list/color_list = list("#FF0000","#FF8C00","#FFFF00","#00FF00","#00BFFF","#0000FF","#9400D3")
+	var/fire_mode = 1
+
+/obj/item/weapon/gun/energy/laser/rainbow/attack_self(mob/living/user as mob)
+
+	switch(fire_mode)
+		if(0)
+			fire_mode = 1
+			to_chat(user, "<span class='warning'>The [src.name] is now set to kill.</span>")
+			projectile_type = "/obj/item/projectile/beam/rainbow"
+			playsound(user,'sound/weapons/egun_toggle_noammo.ogg',73,0,-5)
+			fire_sound = 'sound/weapons/Laser.ogg'
+		if(1)
+			fire_mode = 0
+			to_chat(user, "<span class='warning'>The [src.name] is now set to mindflay.</span>")
+			projectile_type = "/obj/item/projectile/beam/rainbow/braindamage"
+			playsound(user,'sound/weapons/egun_toggle_noammo.ogg',73,0,-5)
+			fire_sound = 'sound/items/quack.ogg'
+
+
+/obj/item/weapon/gun/energy/laser/rainbow/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0, var/use_shooter_turf = FALSE)
+
+	projectile_color = color_list[current_color]
+	if(current_color < 7 )
```

Comment:
Better would be to make it use the length of the `color_list` list, so adding new colours isn't a problem.